lib/custom_set_cases.rb

```diff
@@ -0,0 +1,85 @@
+class CustomSetCase < OpenStruct
+  def name
+    'test_%s' % description.gsub(/ |-/, '_')
+  end
+
+  def test_body
+    send section
+  end
+
+  def union
+    "set1 = CustomSet.new #{set1}
+    set2 = CustomSet.new #{set2}
+    expected = CustomSet.new(#{expected})
+    #{assert_or_refute}_equal expected, set1.union(set2)"
+  end
+
+  def difference
+    "set1 = CustomSet.new #{set1}
+    set2 = CustomSet.new #{set2}
+    expected = CustomSet.new(#{expected})
+    #{assert_or_refute}_equal expected, set1.difference(set2)"
+  end
+
+  def intersection
+    "set1 = CustomSet.new #{set1}
+    set2 = CustomSet.new #{set2}
+    expected = CustomSet.new #{expected}
+    #{assert_or_refute}_equal expected, set2.intersection(set1)"
+  end
+
+  def add
+    "set = CustomSet.new #{set}
+    element = #{element}
+    expected = CustomSet.new #{expected}
+    #{assert_or_refute}_equal expected, set.put(element)"
+  end
+
+  def equal
+    "set1 = CustomSet.new #{set1}
+    set2 = CustomSet.new #{set2}
+    #{assert_or_refute}_equal set1, set2"
+  end
+
+  def disjoint
+    "set1 = CustomSet.new #{set1}
+    set2 = CustomSet.new #{set2}
+    #{assert_or_refute} set1.disjoint? set2"
+  end
+
+  def subset
+    "set1 = CustomSet.new #{set1}
+    set2 = CustomSet.new #{set2}
+    #{assert_or_refute} set2.subset? set1"
+  end
+
+  def empty
+    "set = CustomSet.new #{set}
+    #{assert_or_refute}_equal CustomSet.new, set"
+  end
+
+  def contains
+    "set = CustomSet.new #{set}
+    element = #{element}
+    #{assert_or_refute} set.member? element"
+  end
+
+  def assert_or_refute
+    expected ? 'assert' : 'refute'
+  end
+
+  def skipped
+    index.zero? ? '# skip' : 'skip'
+  end
+end
+
+CustomSetCases = proc do |data|
+  json = JSON.parse(data)
+  cases = []
+  (json.keys - ['#']).each do |section|
+    json[section]['cases'].each_with_index do |row, i|
```

Comment:
I believe this `each_with_index` should be simply an `each` while the above line's `each` should be the `each_with_index`. This is currently leaving the first exercise of each type (diff, union, etc.) unskipped.